query_parser/lib/query_parser.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Proj5.Parser do
+  
```

Comment:
I feel that there's way too much string manipulation in here. The result is complex, and it is only going to get more so.

Could you instead do something like the following (untried) code:

~~~ elixir
  def parse(expression) do
    _parse(expression, _result = [])
  end

  defp _parse("", result), do: result
  
  defp _parse("(" <> rest, result) do
    [ _parse(rest, []) | result ]
  end

  defp _parse(")" <> rest, result) do
    result
  end
  
  defp _parse(<< digit::utf8(1), rest::binary>>, result)
  when digit in ~w{- 0 1 2 3 4 5 6 7 8 9} do
~~~